horace_core/sqw/PixelData/@PixelTmpFileHandler/PixelTmpFileHandler.m

```diff
@@ -89,30 +108,44 @@ function delete_tmp_files(obj)
     function obj = write_float_data_(obj, file_id, pix_data)
         % Write the given data to the file corresponding to the given file ID
         % in float32
-        % TODO: improve this by writing data in chunks to sustain write speeds
-        try
-            fwrite(file_id, pix_data, obj.FILE_DATA_FORMAT_);
-        catch ME
-            switch ME.identifier
-            case 'MATLAB:badfid_mx'
-                error('PIXELTMPFIELHANDLER:write_float_data_', ...
-                  'Could not write to file with ID ''%d'':\n The file is not open', ...
-                  file_id);
-            otherwise
-                tmp_file_path = fopen(file_id);
-                error('PIXELTMPFIELHANDLER:write_float_data_', ...
-                      'Could not write to file ''%s'':\n%s', ...
-                      tmp_file_path, ferror(file_id));
+        SIZE_OF_FLOAT = 4;
+        chunk_idx_length = hor_config().mem_chunk_size/SIZE_OF_FLOAT;
+        data_size = SIZE_OF_FLOAT*numel(pix_data);  % in bytes
+        num_chunks = ceil(data_size/hor_config().mem_chunk_size);
+
+        for i = 1:num_chunks
+            start_idx = (i - 1)*chunk_idx_length + 1;
+            end_idx = min(i*chunk_idx_length, numel(pix_data));
+            try
```

Comment:
the `try ... catch` could be outside the `for` loop to keep the loop chunk smaller. In either case it will abort the loop as soon an an error is thrown, you're not recording the chunk index in any of the log messages.